src/admin.js

```diff
@@ -0,0 +1,431 @@
+/* eslint no-labels: ['error', { 'allowLoop': true }] */
+import 'fetch-everywhere';
+import StitchClient from './client';
+import ADMIN_TYPE from './common';
+
+const v1 = 1;
+const v2 = 2;
+
+export default class Admin extends StitchClient {
+  constructor(baseUrl) {
+    super('', {baseUrl});
+  }
+
+  get type() {
+    return ADMIN_TYPE;
+  }
+
+  get _v2() {
+    const v2do = (url, method, options) =>
+      super._do(
+        url,
+        method,
+        Object.assign({}, {apiVersion: v2}, options)
+      ).then(response => {
+        const contentHeader = response.headers.get('content-type') || '';
+        if (contentHeader.split(',').indexOf('application/json') >= 0) {
+          return response.json();
+        }
+        return response;
+      });
+    return {
+      _get: (url, queryParams) => v2do(url, 'GET', {queryParams}),
+      _put: (url, data) =>
+        (data ?
+          v2do(url, 'PUT', {body: JSON.stringify(data)}) :
+          v2do(url, 'PUT')),
+      _patch: (url, data) =>
+        (data ?
+          v2do(url, 'PATCH', {body: JSON.stringify(data)}) :
+          v2do(url, 'PATCH')),
+      _delete: (url)  => v2do(url, 'DELETE'),
+      _post: (url, body, queryParams) =>
+        (queryParams ?
+          v2do(url, 'POST', { body: JSON.stringify(body), queryParams }) :
+          v2do(url, 'POST', { body: JSON.stringify(body) }))
+    };
+  }
+
+  get _v1() {
+    const v1do = (url, method, options) =>
+      super._do(
+        url,
+        method,
+        Object.assign({}, {apiVersion: v1}, options)
+      ).then(response => response.json());
+    return {
+      _get: (url, queryParams) => v1do(url, 'GET', {queryParams}),
+      _put: (url, options) => v1do(url, 'PUT', options),
+      _delete: (url)  => v1do(url, 'DELETE'),
+      _post: (url, body) => v1do(url, 'POST', {body: JSON.stringify(body)})
+    };
+  }
+
+  _do(url, method, options) {
+    return super._do(url, method, options)
+      .then(response => response.json());
+  }
+
+  profile() {
+    const api = this._v1;
+    return {
+      keys: () => ({
+        list: () => api._get('/profile/keys'),
+        create: (key) => api._post('/profile/keys'),
+        apiKey: (keyId) => ({
+          get: () => api._get(`/profile/keys/${keyId}`),
+          remove: () => api._delete(`/profile/keys/${keyId}`),
+          enable: () => api._put(`/profile/keys/${keyId}/enable`),
+          disable: () => api._put(`/profile/keys/${keyId}/disable`)
+        })
+      })
+    };
+  }
+
+  /**
+   * Ends the session for the current user.
+   *
+   * @returns {Promise}
+   */
+  logout() {
+    return super._do('/auth/session', 'DELETE', { refreshOnFailure: false, useRefreshToken: true, apiVersion: v2 })
+      .then(() => this.auth.clear());
+  }
+
+  /**
+   * Returns profile information for the currently logged in user
+   *
+   * @returns {Promise}
+   */
+  userProfile() {
+    const api = this._v2;
+    return api._get('/auth/profile');
```

Comment:
nit: no purpose in the `const` here really. might as well just do `return this._v2._get(...)`